### PR TITLE
negative test for regular rex run

### PR DESCRIPTION
**Pull request by pondrejk**
### Problem Statement
SAT-28435 (occurs for all regular rex runs, not anisble or pull mode)

### Solution
some negative testing

### CODE REVIEW

**Comment by pondrejk** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py -k  test_negative_run_default_job_template
theforeman:
    smart_proxy_remote_execution_ssh: 119
```

File: tests/foreman/cli/test_remoteexecution.py [hunk 1]
```diff
@@ -134,6 +134,46 @@ def test_positive_run_default_job_template(
...
+    def test_negative_run_default_job_template(
+        self, module_org, rex_contenthost, module_target_sat
+    ):
+        """Confirm that job status is correct if remote action fails
+
+        :id: b6229222-26cf-46bd-bbf6-46814cafd085
+
+        :expectedresults: Verify the job exits as expected
+
+        :verifies: SAT-28435
```
Comment (lhellebr) [inline tests/foreman/cli/test_remoteexecution.py:148]: I hesitate to mark this as regression test for that bug. This is quite a general test that jobs report failure correctly. Actually, I see multiple tests that test this, although the failure is caused by something else (`test_negative_time_to_pickup`, `test_positive_ansible_job_with_nonexisting_module`, `test_positive_run_job_in_chosen_directory`). So I need to ask,
1) Why didn't these tests catch the regression?
2) Isn't it better to fix these tests? Just a thought, I personally think having this general and simple test is a good thing.
3) Shouldn't you write yet another test that reproduces the regression the way it's described in the bug?

**Comment by pondrejk** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py -k  test_negative_run_default_job_template
theforeman:
    smart_proxy_remote_execution_ssh: 119
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8847
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_remoteexecution.py -k  test_negative_run_default_job_template --external-logging
Test Result : ========== 1 failed, 63 deselected, 76 warnings in 1905.82s (0:31:45) ==========
```

File: tests/foreman/cli/test_remoteexecution.py (hunk 1 above)
Comment (pondrejk) [reply tests/foreman/cli/test_remoteexecution.py:148]: The thing is that the issue is not occurring on ansible style rex, nor pull provider rex, that's why the existing negative tests didn't catch it. The bug describes a scenario that uncovered it, but in fact any job failure on client reproduces it, the fix is general as well https://github.com/theforeman/smart_proxy_remote_execution_ssh/pull/119.

**Comment by pondrejk** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py -k  test_negative_run_default_job_template
theforeman:
    smart_proxy_remote_execution_ssh: 119
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8851
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_remoteexecution.py -k  test_negative_run_default_job_template --external-logging
Test Result : ========== 1 passed, 63 deselected, 76 warnings in 2067.96s (0:34:27) ==========
```

File: tests/foreman/cli/test_remoteexecution.py [hunk 2]
```diff
@@ -134,6 +134,50 @@ def test_positive_run_default_job_template(
...
+        ][0]
+        out = module_target_sat.cli.JobInvocation.get_output(
+            {
+                'id': job_id,
+                'host': client.hostname,
+                'organization-id': module_org.id,
+            }
+        )
+        assert 'Exit status: 23' in out
+        assert 'Exit' in out
```
Comment (vsedmik) [inline tests/foreman/cli/test_remoteexecution.py:179]: Is this redundant since `assert 'Exit status: 23' in out` covers it already?

**Review by vsedmik — APPROVED** [review]
LGTM, just one non-blocking nitpick.

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py -k  test_negative_run_default_job_template

**Comment by pondrejk** [comment]
frt, prt is bound to fail on current stream as the pkg is not up to date yet there, executing it anyway for the automerge-cherrypick bot as the pkg should be ok on 6.16

**Pull request merged**